inference_interface.py

```diff
@@ -217,12 +217,17 @@ def numpy_to_toyfile(
                 ds.attrs[k] = dumps(md)
 
 
-def toyfiles_to_numpy(file_name_pattern, numpy_array_names=None):
+def toyfiles_to_numpy(file_name_pattern, numpy_array_names=None,
+                      return_metadata=False):
     filenames = sorted(glob(file_name_pattern))
     dtype_prototype = None
     results = {}
+    metadata = {}
+    array_metadatas = {}
     for fn in filenames:
         with h5py.File(fn, "r") as f:
+            metadata[fn] = {k: v for k, v in f.attrs.items()}
```

Comment:
Since https://github.com/XENONnT/inference_interface/blob/76a71853cae07727464e1cafe635e6a843e51a7c/inference_interface.py#L211 used `dumps`, here it should use `loads`.

Reply:
Otherwise, you will only get strings of dictionaries.